service/api.go

```diff
@@ -130,19 +130,58 @@ func (as *APIServer) FetchTask(w http.ResponseWriter, r *http.Request) {
 	gimlet.WriteJSON(w, t)
 }
 
-// fetchProjectRef returns a project ref given the project identifier
-func (as *APIServer) fetchProjectRef(w http.ResponseWriter, r *http.Request) {
-	id := gimlet.GetVars(r)["identifier"]
-	projectRef, err := model.FindMergedProjectRef(id, "", true)
+// fetchLimitedProjectRef returns a limited project ref given the project identifier
+func (as *APIServer) fetchLimitedProjectRef(w http.ResponseWriter, r *http.Request) {
+	id := gimlet.GetVars(r)["projectId"]
+	p, err := model.FindMergedProjectRef(id, "", true)
 	if err != nil {
 		as.LoggedError(w, r, http.StatusInternalServerError, err)
 		return
 	}
-	if projectRef == nil {
+	if p == nil {
 		http.Error(w, fmt.Sprintf("no project found named '%v'", id), http.StatusNotFound)
 		return
 	}
-	gimlet.WriteJSON(w, projectRef)
+
+	limitedRef := &model.ProjectRef{
+		Id:     p.Id,
+		Owner:  p.Owner,
+		Repo:   p.Repo,
+		Branch: p.Branch,
+		CommitQueue: model.CommitQueueParams{
+			Message: p.CommitQueue.Message,
+			Enabled: p.CommitQueue.Enabled,
+		},
+	}
+
+	gimlet.WriteJSON(w, limitedRef)
+}
+
+// projectWithWorkstationConfig returns a limited project ref with the workstation config given the project identifier
+func (as *APIServer) projectWithWorkstationConfig(w http.ResponseWriter, r *http.Request) {
```

Comment:
Same thing as above, a basic unit test of the functionality would be good here.

Reply:
There is no unit test or a good place to add one, and give that it's deprecated it doesn't feel worth going through the setup. I tested it at http://localhost:9090/api/ref/evergreen though and made sure it looks okay 
<img width="368" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/13104717/ae569664-1045-4ad2-9c35-9423bd1b804c">

Reply:
done